src/modules/connections/connection-manager.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import {ConnectionsRepositoryService} from './connections-repository.service';
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+import {ConnectionData} from './connections.interfaces';
+
+@Injectable()
+export class ConnectionManagerService {
+  private connections: Map<number, Connection> = new Map();
+  constructor(
+    private connectionRepository: ConnectionsRepositoryService,
+  ) {}
+
+  async getConnection(id: number): Promise<Connection> {
+    let connection = this.connections.get(id);
+    try {
+      if (connection && connection.isConnected) {
+        return connection;
+      } else if (connection) {
+        return await connection.connect();
+      }
+      const connectionData: ConnectionData = await this.connectionRepository.getDataForConnectionCreating(id);
+      if (!connectionData) {
+        return null;
+      }
+      connection = await createConnection(connectionData as ConnectionOptions);
+      this.connections.set(id, connection);
+      return connection;
+    } catch (e) {
+      return null;
+    }
+  }
+
+  static async isReachable(connectionData: ConnectionData) {
+    try {
+      const connection = await createConnection(connectionData as ConnectionOptions);
+      if (connection && connection.isConnected) {
+        await connection.close();
+        return true;
+      }
+      return false;
```

Comment:
Should we wait for a connection close?
Mb use without await?

and mb wirte it like
```
const result = connection && connection.isConnected;
connection.close();
return result;

```

Reply:
done